raft/src/main/java/org/apache/kafka/raft/ReplicatedLog.java

```diff
@@ -106,6 +120,14 @@
      */
     void updateHighWatermark(LogOffsetMetadata offsetMetadata);
 
+    /**
+     * Updates the log start offset if necessary.
+     *
+     * The replicated log's start offset can be increased when there is a snapshot greater than the
+     * current log start offset.
+     */
+    boolean updateLogStart(OffsetAndEpoch logStartSnapshotId);
```

Comment:
Perhaps we can make this mirror `truncateFullyToLatestSnapshot`. Perhaps `deleteToLatestSnapshot()` or something like that.

Reply:
Done. Named it `deleteToNewOldestSnapshotId`.